internal/controller/datadogagent/controller_reconcile_dca.go

```diff
@@ -132,3 +134,40 @@ func (r *Reconciler) cleanupV2ClusterAgent(logger logr.Logger, dda *datadoghqv2a
 
 	return reconcile.Result{}, nil
 }
+
+// cleanupOldDCADeployments deletes DCA deployments when a DCA Deployment's name is changed using clusterAgent name override
+func (r *Reconciler) cleanupOldDCADeployments(ctx context.Context, logger logr.Logger, dda *datadoghqv2alpha1.DatadogAgent, resourcesManager feature.ResourceManagers, newStatus *datadoghqv2alpha1.DatadogAgentStatus) error {
+	matchLabels := client.MatchingLabels{
+		apicommon.AgentDeploymentComponentLabelKey: datadoghqv2alpha1.DefaultClusterAgentResourceSuffix,
+		kubernetes.AppKubernetesManageByLabelKey:   "datadog-operator",
+	}
+	deploymentName := getDeploymentNameFromDCA(dda)
+	validDeploymentNames := map[string]struct{}{
```

Comment:
Same as the CCR. We only have one DCA deployment per cluster so we don't need a map for this